exp/api/v1alpha4/azuremanagedcontrolplane_default.go

```diff
@@ -73,3 +73,10 @@ func (r *AzureManagedControlPlane) setDefaultSubnet() {
 		r.Spec.VirtualNetwork.Subnet.CIDRBlock = defaultAKSNodeSubnetCIDR
 	}
 }
+
+// setDefaultAPIServerAccessProfile sets the default APIServerAccessProfile for an AzureManagedControlPlane.
+func (r *AzureManagedControlPlane) setDefaultAPIServerAccessProfile() {
+	if r.Spec.APIServerAccessProfile == nil {
```

Comment:
nit: Is this required? Wouldn't just leaving `nil` also work?

@CecileRobertMichon are there some other examples of setting default values like this?

Reply:
You are right, I don't think this is required. (I was thinking setting it so there will be less working in validation webhook, but i don't think this is a good idea, since it's not good for unit test)